pkg/controller/core.oam.dev/v1beta1/application/application_controller.go

```diff
@@ -256,10 +256,30 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	var phase = common.ApplicationRunning
 	if !hasHealthCheckPolicy(appFile.ParsedPolicies) {
-		app.Status.Services = handler.services
+		for idx, svc := range handler.services {
+			clusters := make(map[string]interface{})
+
+			for _, component := range app.Spec.Components {
+				if _, ok := clusters[svc.Cluster]; component.Name == svc.Name && !ok && svc.Cluster != "" {
+					clusters[svc.Cluster] = struct{}{}
+					isHealthy, message, err := handler.checkComponentHealthWithMessage(ctx, appParser, appFile,
+						component, nil, svc.Cluster, svc.Namespace)
+
+					if err != nil {
+						logCtx.Error(err, "Failed to collect health status")
```

Comment:
Do we need to set the status to fail?

Reply:
Hey, sorry for leaving this PR abandoned for a while. Actually, this was the whole point — to actually update service statuses during reconciliation and set application's state to Failed if some probes completed unsuccessfully. Otherwise the app is eternally 'Healthy' even if some of the deployed services are not (there is an example in #6639).